File: src/main/java/io/github/hotlava03/baclava4j/commands/CommandHandler.java
package io.github.hotlava03.baclava4j.commands;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import discord4j.core.event.domain.message.MessageCreateEvent;

public class CommandHandler {

    private List<Command> commands = new ArrayList<>();

    public List<Command> getCommands() {
        return this.commands;
    }

    public void register(Command... commands) {
        register(Arrays.asList(commands));
    }

    public void register(List<Command> commands) {
        this.commands.addAll(commands);
    }

    public Command getCommandByName(String name) {
        for (Command cmd : commands) {
            if (cmd.getName().equalsIgnoreCase(name))
                return cmd;
            for (String alias : cmd.getAliases())
                if (alias.equalsIgnoreCase(name))
                    return cmd;    
        }
        return null;
    }

    public void execute(MessageCreateEvent e) {
        Command command = this.getCommandByName(e.getMessage().getContent().get().substring(2).split("\\s")[0]);
        command.getExecutor().onCommand(new CommandEvent(e));
    }
}

File: src/main/java/io/github/hotlava03/baclava4j/commands/CommandHandler.kt
package io.github.hotlava03.baclava4j.commands

import discord4j.core.event.domain.message.MessageCreateEvent

class CommandHandler {
    private val commands: MutableList<Command> = ArrayList()
    fun getCommands(): List<Command> {
        return commands
    }

    fun register(vararg commands: Command) {
        register(listOf(*commands))
    }

    fun register(commands: List<Command>) {
        this.commands.addAll(commands)
    }

    fun getCommandByName(name: String): Command? {
        for (cmd in commands) {
            if (cmd.name.equals(name, ignoreCase = true)) return cmd
            for (alias in cmd.aliases ?: arrayOf()) if (alias.equals(name, ignoreCase = true)) return cmd
        }
        return null
    }

    fun execute(e: MessageCreateEvent) {
        val command = getCommandByName(e.message.content.get().substring(2).split("\\s").toTypedArray()[0])!!
        command.executor.onCommand(CommandEvent(e))
    }
}